### PR TITLE
Add error pages and routes

### PR DESCRIPTION
**Pull request by jparkrr**
Closes partially #412 

I added some docs here https://github.com/MoveOnOrg/mop-frontend/blob/5832a82a8a1c04f66d5071c5fdfb446680cbecf9/docs/EXPLANATION--error-handling.md

- [x] Add legacy error pages
- [x] Add giraffe error pages
- [x] Show error if window.error is set
- [x] Show error on loading petition page with `fetch()` error
  - [x] Server doesn't respond (user could be offline)
  - [x] Server returns error status code
- [ ] Show error on signing petition with `fetch()` error
  - [ ] Server doesn't respond (user could be offline)
  - [ ] Server returns error status code
- [x] Show error on static pages with `fetch()` error
  - [x] Server doesn't respond (user could be offline)
  - [x] Server returns error status code

### CODE REVIEW

File: src/components/theme-giraffe/error404.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <div className='row my-5'>
+      <div className='col-12'>
+        <h1 className='text-align-center'>
+          Oops — looks like what you’re looking for isn’t here!
+        </h1>
+        <p className='text-align-center'>
+          {description}
+          {!description && (
+            <span>
+              Email us at <a href='mailto:help@moveon.org'>help@moveon.org</a>{' '}
```
Comment (schuyler1d) [inline src/components/theme-giraffe/error404.js]: ha ha!  I love this pattern `{' '}` to get around react removing white space.

File: src/components/theme-giraffe/error404.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+              so we can help you find your way.
+            </span>
+          )}
+        </p>
+        <img
+          src='https://static.moveon.org/giraffe/images/error404.svg'
```
Comment (schuyler1d) [inline src/components/theme-giraffe/error404.js]: This should use STATIC_ROOT

File: src/components/theme-giraffe/error500.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+              We’re sorry about that! Please go back and try again. If you’re
+              still having trouble, shoot us a note at{' '}
+              <a href='mailto:help@moveon.org'>help@moveon.org</a>.
+            </span>
+          )}
+        </p>
+      </div>
+      <div className='col-12 col-md-5 order-md-first'>
+        <img
+          src='https://static.moveon.org/giraffe/images/error500.svg'
```
Comment (schuyler1d) [inline src/components/theme-giraffe/error500.js]: STATIC_ROOT

File: src/components/theme-giraffe/error500.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export const Error500 = ({ error: { description } }) => (
```
Comment (schuyler1d) [inline src/components/theme-giraffe/error500.js]: Can you put some documentation about the 'error arc' in this file/page -- I feel like I'm going to want to understand how it all shows up or doesn't and this seems like a good place to look for it in the code. -- basically how does SERVER_ERROR get seen, and then the wrapper story of when it displays.

**Comment by jparkrr** [comment]
@schuyler1d Really good point about the '*' route. It would match for every route and we would no longer have the intended behavior of redirecting to `${PROD_URL}${path}` if the component isn't implemented. Implemented routes not marked `prodReady` would have redirected correctly.

It would ALSO have broken where we do appLocation.push('https://....')

So I will remove the fallback route. Thanks!

**Pull request merged**